lib/grant-types/authorization-code-grant-type.js

```diff
@@ -176,7 +176,7 @@ AuthorizationCodeGrantType.prototype.revokeAuthorizationCode = function(code) {
 
 AuthorizationCodeGrantType.prototype.saveToken = function(user, client, authorizationCode, scope) {
   const fns = [
-    this.validateScope(user, client, scope),
+    scope ? this.validateScope(user, client, scope) : scope,
     this.generateAccessToken(client, user, scope),
```

Comment:
These would still be generated if scope is invalid

Reply:
It should be moved outside the promise.all to resolve #84

Reply:
So if scope is not null, undefined or an empty string it will be validated, but if it is null or undefined or empty string it will be passed on. 

I also think it should be outside the array, as Promise.all does not have a revert functionality, if validateScope is throwing or returning undefined/empty string. Rename scope from line 177 to parmScope/requestedScope and see that we would not pass the scope from the fns-Array to the other functions but take the unvalidated scope from the parameters.